crates/biome_js_analyze/src/lint/suspicious/use_iterable_callback_return.rs

```diff
@@ -80,6 +80,23 @@ declare_lint_rule! {
     /// ```js
     /// [].forEach(() => void null); // Void return value, which doesn't trigger the rule
     /// ```
+    ///
+    /// ## Options
+    ///
+    /// ### `checkForEach`
+    ///
+    /// Default: `true`
+    ///
+    /// When set to `false`, rule will skip reporting `forEach` callbacks that return a value.
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "checkForEach": true
+    ///     }
+    /// }
+    /// ```
```

Comment:
Please check the contribution guide for how the format of options should be.

Since the default is `true`, there's little to no value showing an example with `true`. Instead, we should have an option block with `false`, and an example (with or without diagnostics) that shows how `false` works in that snippet/example.

Even if we change the default value of the option, the suggestion is still valid. We don't want to test the defaults, but the other values